libbess-python/bess_test.py

```diff
@@ -0,0 +1,106 @@
+import unittest
+import bess_msg_pb2 as bess_msg
+import error_pb2 as error_msg
+import service_pb2
+import grpc
+import proto_conv
+import bess
+import time
+from concurrent import futures
+
+class TestServiceImpl(service_pb2.BESSControlServicer):
+    def __init__(self):
+        pass
+
+    def KillBess(self, request, context):
+        return bess_msg.EmptyResponse()
+
+    def CreatePort(self, request, context):
+        response = bess_msg.CreatePortResponse()
+        response.name = request.name
+        return response
+
+    def ModuleCommand(self, request, context):
+        response = bess_msg.ModuleCommandResponse()
+        return response
+
+    def ListModules(self, request, context):
+        response = bess_msg.ListModulesResponse()
+        return response
+
+class TestBESS(unittest.TestCase):
+    def setUp(self):
+        self.server = grpc.server(futures.ThreadPoolExecutor(max_workers=2))
+        service_pb2.add_BESSControlServicer_to_server(
+            TestServiceImpl(),
+            self.server)
+        self.server.add_insecure_port('[::]:%d' % bess.BESS.DEF_PORT)
+        self.server.start()
+
+    def tearDown(self):
+        self.server.stop(0)
+
+    def test_connect(self):
+        client = bess.BESS()
+        client.connect()
+        time.sleep(1)
+        self.assertEqual(True, client.is_connected())
+
+        client.disconnect()
+        time.sleep(1)
+        self.assertEqual(False, client.is_connected())
+
+    def test_kill(self):
+        client = bess.BESS()
+        client.connect()
+
+        response = client.kill()
+        self.assertEqual(0, response.error.err)
+
+    def test_list_modules(self):
+        client = bess.BESS()
+        client.connect()
+
+        response = client.list_modules()
+        self.assertEqual(0, response.error.err)
+
+    def test_create_port(self):
+        client = bess.BESS()
+        client.connect()
+
+        response = client.create_port('pcap_port', 'p0', {'dev': 'rnd'})
+        self.assertEqual(0, response.error.err)
+        self.assertEqual('p0', response.name)
+
+        response = client.create_port('pmd_port', 'p0', {
+            'loopback': True,
+            'port_id': 14325,
+            'pci': 'akshdkashf'})
+        self.assertEqual(0, response.error.err)
+        self.assertEqual('p0', response.name)
+
+        response = client.create_port('unix_port', 'p0', {'path': '/ajksd/dd'})
+        self.assertEqual(0, response.error.err)
+        self.assertEqual('p0', response.name)
+
+        response = client.create_port('vport', 'p0', {
+            'ifname': 'veth0',
+            'container_pid': 23124,
+            'rxq_cpus': [1,2,3],
+            'tx_tci': 123,
+            'tx_outer_tci':123,
+            'loopback': False,
+            'ip_addrs': ['1.1.1.1', '2.2.2.2']
```

Comment:
IP addresses that are not palindrome would be better, to test if endianness works correctly